apps/prairielearn/src/lib/instructorFiles.js

```diff
@@ -163,5 +166,23 @@ export function getPaths(req, res, callback) {
       }
     });
 
+  return paths;
+}
+
+/**
+ * Wrapper around {@link getPaths} to support callback-based usage.
+ *
+ * @param {import('express').Request} req
+ * @param {import('express').Response} res
+ * @param {(err: Error | null | undefined, paths?: any) => void} callback
+ */
+export function getPathsCallback(req, res, callback) {
```

Comment:
Is there a reason why `util.callbackify` wasn't used? I can't find anything that is specific to this case that wouldn't handle this problem.

Reply:
`util.callbackify` is only meant for async functions: https://nodejs.org/api/util.html#utilcallbackifyoriginal

You can see this in the implementation: https://github.com/nodejs/node/blob/d96606d9aab35baf9e5c77cf9312145c3ae6086b/lib/util.js#L332

Or prove it to yourself with this little snippet:

```js
const util = require('util');

const callbackFoo = util.callbackify(() => 'hello');

callbackFoo((err, val) => console.log(err, val));
```

Rather than seeing `null 'hello'` printed, it errors in Node internals:

```
node:util:313
      .then((ret) => process.nextTick(cb, null, ret),
       ^

TypeError: ReflectApply(...).then is not a function
    at Callbackified (node:util:313:8)
    at Object.<anonymous> (/Users/nathan/git/PrairieLearn/test.js:5:1)
    at Module._compile (node:internal/modules/cjs/loader:1241:14)
    at Module._extensions..js (node:internal/modules/cjs/loader:1295:10)
    at Module.load (node:internal/modules/cjs/loader:1091:32)
    at Module._load (node:internal/modules/cjs/loader:938:12)
    at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:83:12)
    at node:internal/main/run_main_module:23:47
```